### PR TITLE
add PATH de-duplication when searching for plugins

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://github.com/kubernetes/kubectl/issues/575

cc @ahmetb @soltysh 

```release-note
NONE
```

/sig cli
/kind bug

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/kind bug

File: pkg/kubectl/cmd/plugin/plugin.go [hunk 1]
```diff
@@ -238,3 +242,18 @@ func isExecutable(fullPath string) (bool, error) {
...
+// sorting or otherwise altering its order in any way.
+func uniquePathsList(paths []string) []string {
+	seen := map[string]bool{}
+	newPaths := []string{}
+	for _, p := range paths {
+		if seen[p] {
+			continue
+		}
+		seen[p] = true
+		newPaths = append(newPaths, p)
```
Comment (tedyu) [inline pkg/kubectl/cmd/plugin/plugin.go]: How about accumulating the count of unique paths in the loop and allocate array, newPaths, outside the loop ?

This way there is no need to call append which saves some memory.

Comment (soltysh) [reply pkg/kubectl/cmd/plugin/plugin.go]: I don't think that matters that much here. You'd have to iterate over the paths array twice in that case.

**Pull request merged**